### PR TITLE
add workspace cleanup action

### PR DESCRIPTION
**Pull request by artembo**
It will be used for cleaning up workspace in self-hosted runnsers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Tarantool Actions
+
+A repository with commonly used GitHub actions used in
```
Comment (ylobankov) [inline README.md:3]: `... used ... used ...` - probably we can fix it so:

A repository with commonly used GitHub actions for the `tarantool` organization on GitHub.

File: cleanup/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Setup environment
+
+Action cleans up workspace directory after previous run. The main reason to add
+this action is [tarantool/tarantool-qa#145](https://github.com/tarantool/tarantool-qa/issues/145). 
+When submodules have changed, the standard checkout fails with git errors.
```
Comment (ylobankov) [inline cleanup/README.md:5]: have -> are

File: cleanup/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Setup environment
+
+Action cleans up workspace directory after previous run. The main reason to add
+this action is [tarantool/tarantool-qa#145](https://github.com/tarantool/tarantool-qa/issues/145). 
+When submodules have changed, the standard checkout fails with git errors. 
+The actions/checkout error still opened
```
Comment (ylobankov) [inline cleanup/README.md:6]: 1. Please wrap `actions/checkout` into backtick symbols.
2. `error still opened` -> `error is still opened`

File: cleanup/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+[actions/checkout#418](https://github.com/actions/checkout/issues/418)
+
+## How to use GitHub Action from GitHub workflow
+
+Add the following code to the running steps after checkout done:
```
Comment (ylobankov) [inline cleanup/README.md:11]: after?

File: cleanup/action.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+name: 'Clean workspace'
+description: 'Remove workspace directory after previous workflow run'
+runs:
+  using: "composite"
```
Comment (ylobankov) [inline cleanup/action.yml:4]: Please use single quotes for consistency.

File: cleanup/action.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+name: 'Clean workspace'
+description: 'Remove workspace directory after previous workflow run'
```
Comment (ylobankov) [inline cleanup/action.yml:2]: Remove -> Clean

File: cleanup/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Setup environment
+
+Action cleans up workspace directory after previous run. The main reason to add
```
Comment (ylobankov) [inline cleanup/README.md:3]: In other places you don't use `up`. Either to remove it from here or add to everywhere.

File: cleanup/action.yml (hunk 6 above)
Comment (artembo) [reply cleanup/action.yml:2]: done

File: cleanup/README.md (hunk 7 above)
Comment (artembo) [reply cleanup/README.md:3]: done

File: cleanup/README.md (hunk 4 above)
Comment (artembo) [reply cleanup/README.md:11]: before

File: cleanup/README.md (hunk 3 above)
Comment (artembo) [reply cleanup/README.md:6]: done

File: cleanup/README.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Setup environment
+
+Action cleans workspace directory after previous run. The main reason to add
+this action is [tarantool/tarantool-qa#145](https://github.com/tarantool/tarantool-qa/issues/145). 
+When submodules are changed, the standard checkout fails with git errors. 
+The `actions/checkout` error is still opened
+[actions/checkout#418](https://github.com/actions/checkout/issues/418)
```
[the remark this answers is not in the document]
Comment (artembo) [reply cleanup/README.md]: done

File: cleanup/action.yml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+name: 'Clean workspace'
+description: 'Clean workspace directory after previous workflow run'
+runs:
+  using: 'composite'
+  steps:
+    - run: |
+        set -e
+        shopt -s dotglob
+        rm -rf *
```
[the remark this answers is not in the document]
Comment (artembo) [reply cleanup/action.yml]: done

**Review by ylobankov — APPROVED** [review]
LGTM

**Pull request merged**